### DISCUSSION TITLE
Suggestion to Cleanup  / remove older preview tags

### DISCUSSION THREAD

**Discussion by Josverl** [micropython/micropython#16206] [Core Development]
In my experience the switch to use the `preview` tags has made is simpler to understand the difference between latest/stable , specific versions and the (latest) preview.

But now that we are in the 4th version that uses the preview tag, I wonder if there is any value in keeping the older/non-current preview tags.
or if they can better be removed once the actual corresponding release is made.

![image](https://github.com/user-attachments/assets/0ab1bbc9-e9f7-4cab-95a8-b904438cc268)

From an operational point of view I have already added helper functions to use the "most recent preview" whenever I need "the preview" version,
But I would favor having the tags mirror reality in that there is only a single preview version.

In my understanding each of the 'v<older>-preview` tags point to a commit just before the next version , and I cant see any practical use for them, and sometimes, though not often , they lead to misunderstanding
Could these older tags be removed , and also add that action to the ""release runbook" that I assume exists somewhere 

- v1.22.0-preview 
- v1.23.0-preview
- v1.24.0-preview

**Comment by bidrohini** [comment]
You can implement a policy to delete older vX.Y.Z-preview tags once the corresponding stable version (vX.Y.Z) is released.

**Reply by Josverl** [reply]
Care to explain that 'policy' in some more detail?

The only options I'm aware of is a combination of a GH Action and a custom script.
